### PR TITLE
Added laravel pulse | Country list api | Change user status api

### PR DESCRIPTION
**Pull request by dhrumin-7span**
In this PR :

-  Added laravel pulse
-  Added country seeder
-  Country list and user status update API

### CODE REVIEW

File: app/Http/Controllers/Api/V1/Admin/UserStatusController.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+use App\Http\Controllers\Controller;
+use App\Http\Requests\User\ChangeStatus;
+
+class UserStatusController extends Controller
+{
+    use ApiResponser;
+
+    private $userService;
+
+    public function __construct(UserService $userService)
```
Comment (harshil-7span) [inline app/Http/Controllers/Api/V1/Admin/UserStatusController.php:18]: @dhrumin-7span No need take the object in the function param

File: app/Http/Controllers/Api/V1/CountryController.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Country\Index;
+use App\Http\Resources\Country\Collection as CountryCollection;
+
+class CountryController extends Controller
+{
+    use ApiResponser;
+
+    public function __construct(private CountryService $countryService)
+    {
```
Comment (harshil-7span) [inline app/Http/Controllers/Api/V1/CountryController.php:17]: @dhrumin-7span Make it same as UserStatusChangeController

File: app/Http/Controllers/Api/V1/CountryController.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            ),
+        ],
+        responses: [
+            new OA\Response(
+                response: 200,
+                description: 'Success'
+            ),
+        ],
+    )]
+    public function index(Index $request)
```
Comment (harshil-7span) [inline app/Http/Controllers/Api/V1/CountryController.php:69]: @dhrumin-7span Remove the request file

File: app/Http/Requests/Country/Index.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'limit' => 'nullable',
```
Comment (harshil-7span) [inline app/Http/Requests/Country/Index.php:25]: @dhrumin-7span Remove it

File: app/Models/Country.php [hunk 5]
```diff
@@ -2,14 +2,25 @@
...
+use App\Traits\BaseModel;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 
 class Country extends Model
 {
-    use HasFactory;
+    use BaseModel,HasFactory;
+
+    public $queryable = [
```
Comment (harshil-7span) [inline app/Models/Country.php:13]: @dhrumin-7span Please remove it

File: app/Models/Country.php [hunk 6]
```diff
@@ -2,14 +2,25 @@
...
     protected $fillable = [
-        'country_code',
+        'iso',
+        'name',
+        'iso3',
+        'numcode',
+        'phonecode',
     ];
+
+    protected $relationship = [];
```
Comment (harshil-7span) [inline app/Models/Country.php:25]: @dhrumin-7span Please remove the empty relationship varibale

File: app/Services/UserService.php [hunk 7]
```diff
@@ -23,31 +28,26 @@ public function resource(int $id, array $inputs = []): User
 
     public function update(int $id, array $inputs = []): array
     {
-        $user = User::find(auth()->id());
+        $user = $this->resource($id);
```
Comment (harshil-7span) [inline app/Services/UserService.php:31]: @dhrumin-7span Please use the Auth::user();

File: app/Services/UserService.php [hunk 8]
```diff
@@ -56,4 +56,14 @@ public function update(int $id, array $inputs = []): array
 
         return $data;
     }
+
+    public function changeStatus(object $user, array $inputs = [])
+    {
+        $user->update($inputs);
+        $data = [
+            'message' => __('message.changeStatusSuccess', ['status' => $inputs['status']]),
```
Comment (harshil-7span) [inline app/Services/UserService.php:64]: @dhrumin-7span Please add this message in the entity file

File: bootstrap/app.php [hunk 9]
```diff
@@ -16,6 +16,10 @@
                 ->prefix('api/v1')
                 ->group(base_path('routes/api-v1.php'));
 
+            Route::middleware('api')
+                ->prefix('api/v1/admin')
```
Comment (harshil-7span) [inline bootstrap/app.php:20]: @dhrumin-7span The admin API file name shoud be admin-v1.php

File: app/Http/Controllers/Api/V1/Admin/UserStatusController.php (hunk 1 above)
Comment (dhrumin-7span) [reply app/Http/Controllers/Api/V1/Admin/UserStatusController.php:18]: Removed.

File: app/Http/Controllers/Api/V1/Admin/UserStatusController.php [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        ],
+        security: [[
+            'bearerAuth' => [],
+        ]],
+    )]
+    public function __invoke(User $user, ChangeStatus $request)
+    {
+        $user = $this->userService->changeStatus($user, $request->validated());
+
+        return $this->success($user, 200);
```
[the remark this answers is not in the document]
Comment (dhrumin-7span) [reply app/Http/Controllers/Api/V1/Admin/UserStatusController.php:78]: Removed.

File: app/Http/Controllers/Api/V1/CountryController.php (hunk 2 above)
Comment (dhrumin-7span) [reply app/Http/Controllers/Api/V1/CountryController.php:17]: Changed according to UserStatusController.

File: app/Http/Controllers/Api/V1/CountryController.php (hunk 3 above)
Comment (dhrumin-7span) [reply app/Http/Controllers/Api/V1/CountryController.php:69]: Removed.

File: app/Models/Country.php (hunk 5 above)
Comment (dhrumin-7span) [reply app/Models/Country.php:13]: Removed.

File: app/Models/Country.php (hunk 6 above)
Comment (dhrumin-7span) [reply app/Models/Country.php:25]: Removed.

File: app/Services/AuthService.php [hunk 11]
```diff
@@ -121,6 +121,10 @@ public function login(array $inputs): array
             throw new CustomException(__('auth.failed'));
         }
 
+        if ($user->status == config('site.user_status.inactive')) {
+            throw new CustomException(__('message.userInactiveFound'));
```
[the remark this answers is not in the document]
Comment (dhrumin-7span) [reply app/Services/AuthService.php:125]: Changed variable name.

File: bootstrap/app.php (hunk 9 above)
Comment (dhrumin-7span) [reply bootstrap/app.php:20]: Changed admin api file name.

File: app/Services/UserService.php (hunk 8 above)
Comment (dhrumin-7span) [reply app/Services/UserService.php:64]: Now, This message is from the entity file.

File: app/Services/UserService.php (hunk 7 above)
Comment (dhrumin-7span) [reply app/Services/UserService.php:31]: Changed.

File: app/Services/UserService.php [hunk 12]
```diff
@@ -23,31 +29,26 @@ public function resource(int $id, array $inputs = []): User
 
     public function update(int $id, array $inputs = []): array
     {
-        $user = User::find(auth()->id());
+        $user = $this->userObj->find(Auth::id());
```
Comment (harshil-7span) [inline app/Services/UserService.php:32]: @dhrumin-7span Please use the Auth::user()

File: database/migrations/2024_06_19_072618_create_countries_table.php [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('countries', function (Blueprint $table) {
```
Comment (harshil-7span) [inline database/migrations/2024_06_19_072618_create_countries_table.php:11]: @dhrumin-7span Please add the index on the below fields
- name
- phonecode
- numcode

File: routes/api-v1.php [hunk 14]
```diff
@@ -22,3 +23,4 @@
 });
 
 Route::post('generate-signed-url', SignedUrlController::class);
+Route::apiResource('countries', CountryController::class)->only('index');
```
Comment (harshil-7span) [inline routes/api-v1.php:26]: @dhrumin-7span Please define get route instead of apiResource

Comment (dhrumin-7span) [reply routes/api-v1.php:26]: Done

File: database/migrations/2024_06_19_072618_create_countries_table.php (hunk 13 above)
Comment (dhrumin-7span) [reply database/migrations/2024_06_19_072618_create_countries_table.php:11]: Added indexing on this fields.

File: app/Services/UserService.php (hunk 12 above)
Comment (dhrumin-7span) [reply app/Services/UserService.php:32]: Changed.

**Review by harshil-7span — APPROVED** [review]

**Pull request merged**